lib/utils.js

```diff
@@ -13,6 +13,25 @@ module.exports = {
     });
   },
 
+	getApiGatewayStageName() {
+		let stageNameStage, stageNameDeploy;
+		const serviceTemplate = this.serverless.service.resources;
+		Object.keys(serviceTemplate.Resources).forEach(function(key){
+			if (serviceTemplate.Resources[key]['Type'] == 'AWS::ApiGateway::Stage') {
+				stageNameStage = serviceTemplate.Resources[key].Properties.StageName;
```

Comment:
the one being deployed by ApiGatewayDeployement is the unused stage, not the stage created by AWS::ApiGateway::Stage.
it will firstly deploy apig by the  ApiGatewayDeployement with an unused stage, then the AWS::ApiGateway::Stage create a new stage and attach to ApiGatewayDeployement